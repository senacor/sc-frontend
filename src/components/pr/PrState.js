/*
 *
 * DEPRECATED
 *
 */
import React from 'react';
import { connect } from 'react-redux';
import Typography from '@material-ui/core/Typography';
import ListItem from '@material-ui/core/ListItem';
import List from '@material-ui/core/List';
import ListItemText from '@material-ui/core/ListItemText';
import Stepper from '@material-ui/core/Stepper';
import Step from '@material-ui/core/Step';
import StepLabel from '@material-ui/core/StepLabel';
import StepContent from '@material-ui/core/StepContent';
import Button from '@material-ui/core/Button';
import Paper from '@material-ui/core/Paper';
import Icon from '@material-ui/core/Icon';
import Grid from '@material-ui/core/Grid';
import { Link } from 'react-router-dom';
import ObjectGet from 'object-get';

import { withStyles } from '@material-ui/core/styles/index';
import { isEmployee, isSupervisor } from '../../helper/checkRole';
import { createSelector } from 'reselect';
import { getPrDetail, getUserroles } from '../../reducers/selector';
import * as actions from '../../actions';
import { translationMap } from '../translate/Translate';
import { prStatusEnum } from '../../helper/prStatus';

const styles = theme => ({
  paper: {
    backgroundColor: 'inherit'
  },
  stepper: {
    backgroundColor: 'inherit',
    padding: '0',
    width: '100%'
  },
  typography: {
    color: '#FFF',
    marginLeft: '30px',
    marginTop: '20px',
    fontSize: '15px'
  },
  buttonDesktop: {
    position: 'relative',
    marginRight: '1%',
    backgroundColor: theme.palette.primary['400'],
    color: '#FFF',
    marginBottom: '2%'
  },
  buttonDesktopSchedulingDone: {
    position: 'relative',
    marginRight: '1%',
    backgroundColor: theme.palette.primary['A700'],
    color: '#FFF',
    marginBottom: '2%'
  }
});

const progressStructure = [
  {
    text: 'Vorbereitung PR',
    statuses: [
      prStatusEnum.RELEASED_SHEET_EMPLOYEE,
      prStatusEnum.RELEASED_SHEET_REVIEWER,
      prStatusEnum.FIXED_DATE
    ]
  },
  {
    text: 'PR abschließen (Beurteiler)',
    statuses: [prStatusEnum.FINALIZED_REVIEWER]
  },
  {
    text: 'PR abschließen (Mitarbeiter)',
    statuses: [prStatusEnum.FINALIZED_EMPLOYEE]
  },
  {
    text: 'PR archivieren (HR)',
    statuses: [prStatusEnum.ARCHIVED_HR]
  }
];

function isDone(prStatuses, status) {
  if (prStatuses === undefined) {
    return false;
  } else {
    return prStatuses.includes(status);
  }
}

function mainStepIsDone(prStatusesDone, stepId) {
  if (prStatusesDone !== undefined) {
    let finishedSubTasks = progressStructure[stepId].statuses.map(
      status => prStatusesDone[status]
    );
    if (!finishedSubTasks.includes(undefined)) {
      return !finishedSubTasks.includes(false);
    } else {
      return false;
    }
  } else {
    return false;
  }
}

export const getFinishedMilestones = () => {
  return createSelector(
    [state1 => state1.prs, state2 => state2.prDetailId],
    (prs, prDetailId) => {
      if (prs[prDetailId] === undefined) {
        return undefined;
      } else {
        let prStatuses = ObjectGet(prs[prDetailId], 'statuses');
        let isDoneStatusMap = {};
        for (let status in prStatusEnum) {
          isDoneStatusMap[prStatusEnum[status]] = isDone(
            prStatuses,
            prStatusEnum[status]
          );
        }
        return isDoneStatusMap;
      }
    }
  );
};

class PrState extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  changeStateOnButtonClick = status => event => {
    let pr = this.props.prById;
    if (!event.disabled) {
      this.props.addPrStatus(pr, status);
    }
  };

  getExtraStepContent(status) {
    let isStatusDoneMap = this.props.prStatusesDone;
    let forEmployee = isEmployee(this.props.userroles);
    let forReviewer = isSupervisor(this.props.userroles);
    let isReleasedByEmployee = status === prStatusEnum.RELEASED_SHEET_EMPLOYEE;
    let isReleasedByReviewer = status === prStatusEnum.RELEASED_SHEET_REVIEWER;
    let supervisorFinalizesPr = status === prStatusEnum.FINALIZED_REVIEWER;
    let employeeFinalizesPr = status === prStatusEnum.FINALIZED_EMPLOYEE;
    let meetingIsScheduled = status === prStatusEnum.FIXED_DATE;
    let done =
      isStatusDoneMap === undefined
        ? false
        : ObjectGet(isStatusDoneMap, status);
    let { classes } = this.props;
    switch (status) {
      case prStatusEnum.RELEASED_SHEET_EMPLOYEE:
      case prStatusEnum.RELEASED_SHEET_REVIEWER:
      case prStatusEnum.FIXED_DATE:
        return (
          <Grid
            key={`SubStepGrid_${status}`}
            container
            alignItems="center"
            spacing={8}
          >
            <Grid item xl={6} lg={12} md={12} sm={12} xs={12}>
              <Grid container direction="row">
                <Grid item lg={8} md={6} sm={8} xs={8}>
                  <Typography>{translationMap[status]}</Typography>
                </Grid>
                <Grid item lg={4} md={6} sm={4} xs={4}>
                  {done ? <Icon>done</Icon> : null}
                </Grid>
              </Grid>
            </Grid>
            {(forEmployee && isReleasedByEmployee) ||
            (!forEmployee && isReleasedByReviewer) ? (
              <Grid item xl={6} lg={12} md={12} sm={12} xs={12}>
                <Button
                  disabled={done}
                  className={
                    done ? classes.buttonDesktopDisabled : classes.buttonDesktop
                  }
                  onClick={this.changeStateOnButtonClick(status)}
                >
                  PR freigeben
                </Button>
              </Grid>
            ) : null}
            {forEmployee && meetingIsScheduled ? (
              <Grid item xl={6} lg={12} md={12} sm={12} xs={12}>
                <Link
                  to={`/prs/${this.props.prById.id}/scheduling`}
                  style={{ textDecoration: 'none' }}
                >
                  <Button
                    id="scheduling"
                    className={
                      done
                        ? classes.buttonDesktopSchedulingDone
                        : classes.buttonDesktop
                    }
                  >
                    {done ? 'Termindetails' : 'Termin finden'}
                  </Button>
                </Link>
              </Grid>
            ) : null}
          </Grid>
        );
      case prStatusEnum.FINALIZED_REVIEWER:
        return (
          <div key={`SubStepGrid_${status}`}>
            {!forEmployee && supervisorFinalizesPr ? (
              <Button
                disabled={done}
                className={
                  done ? classes.buttonDesktopDisabled : classes.buttonDesktop
                }
                onClick={this.changeStateOnButtonClick(status)}
              >
                Abschließen
              </Button>
            ) : null}
          </div>
        );
      case prStatusEnum.FINALIZED_EMPLOYEE:
        return (
          <div key={`SubStepGrid_${status}`}>
            {!forReviewer && employeeFinalizesPr ? (
              <Button
                disabled={done}
                className={
                  done ? classes.buttonDesktopDisabled : classes.buttonDesktop
                }
                onClick={this.changeStateOnButtonClick(status)}
              >
                Abschließen
              </Button>
            ) : null}
          </div>
        );
      default:
        return null;
    }
  }

  render() {
    const { classes } = this.props;

    return (
      <Paper className={classes.paper}>
        <List>
          <ListItem>
            <ListItemText>Fortschritt des Reviewprozesses</ListItemText>
          </ListItem>
          <ListItem>
            <Stepper
              orientation="vertical"
              className={classes.stepper}
              activeStep={
                [...Array(progressStructure.length).keys()].filter(stepId =>
                  mainStepIsDone(this.props.prStatusesDone, stepId)
                ).length
              }
              onChange={() => {}}
            >
              {progressStructure.map(progressStep => {
                return (
                  <Step key={progressStep.text}>
                    <StepLabel>{progressStep.text}</StepLabel>
                    <StepContent TransitionProps={{ in: true }}>
                      {progressStep.statuses.map(status => {
                        return this.getExtraStepContent(status);
                      })}
                    </StepContent>
                  </Step>
                );
              })}
            </Stepper>
          </ListItem>
        </List>
      </Paper>
    );
  }
}

export const StyledComponent = withStyles(styles)(PrState);
export default connect(
  state => ({
    prById: getPrDetail()(state),
    prStatusesDone: getFinishedMilestones()(state),
    userroles: getUserroles(state)
  }),
  {
    addPrStatus: actions.addPrStatus
  }
)(StyledComponent);
