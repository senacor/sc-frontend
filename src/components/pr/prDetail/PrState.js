import React from 'react';
import { connect } from 'react-redux';
import ListItem from '@material-ui/core/ListItem';
import List from '@material-ui/core/List';
import ListItemText from '@material-ui/core/ListItemText';
import Paper from '@material-ui/core/Paper';

import { withStyles } from '@material-ui/core/styles/index';
import {
  getPrDetail,
  getUserroles,
  getUserinfo,
  getRequiredFields,
  getPrRatings,
  getPrEmployeeContributions,
  getMeeting,
  getSavingThreads
} from '../../../reducers/selector';
import * as actions from '../../../actions';
import { prStatusEnum } from '../../../helper/prStatus';
import PrStatusActionButton from './PrStatusActionButton';
import PrStatusStepper from './PrStateStepper';
import { isHr } from '../../../helper/checkRole';
import { changeRequiredFields } from '../../../actions/sheet';
import { hasRoleInPrBasedOnUserName } from '../../../helper/hasRoleInPr';
import { CheckRequiredClick } from '../../hoc/CheckRequiredClick';
import Typography from '@material-ui/core/Typography';
import { injectIntl } from 'react-intl';

const styles = theme => ({
  paper: {
    backgroundColor: 'inherit',
    marginBottom: 2 * theme.spacing.unit
  },
  typography: {
    color: '#FFF',
    marginLeft: '30px',
    marginTop: '20px',
    fontSize: '15px'
  },
  buttonDesktop: {
    position: 'relative',
    marginRight: '1%',
    backgroundColor: theme.palette.primary['400'],
    color: '#FFF',
    marginBottom: '2%'
  },
  buttonDesktopSchedulingDone: {
    position: 'relative',
    marginRight: '1%',
    backgroundColor: theme.palette.primary['A700'],
    color: '#FFF',
    marginBottom: '2%'
  }
});

class PrState extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  mainStepIsDone = (prStatusesDone, stepId, stepStructure) => {
    if (prStatusesDone !== undefined) {
      return Object.values(stepStructure[stepId].substeps).every(
        substep => substep.isCompleted
      );
    }
    return false;
  };

  calculateActiveStep = (prStatusesDone, stepStructure) => {
    return [...Array(stepStructure.length).keys()].filter(stepId =>
      this.mainStepIsDone(prStatusesDone, stepId, stepStructure)
    ).length;
  };

  getCompletedSubsteps = prStatuses => {
    let isDoneStatusMap = {};
    for (let status in prStatusEnum) {
      isDoneStatusMap[prStatusEnum[status]] = this.isDone(
        prStatuses,
        prStatusEnum[status]
      );
    }
    return isDoneStatusMap;
  };

  isDone = (prStatuses, status) => {
    if (prStatuses === undefined) {
      return false;
    } else {
      return prStatuses.includes(status);
    }
  };

  updateStepStructure = (pr, userinfo, prStatusesDone, meeting) => {
    let userIsMemberOfHr = isHr(this.props.userroles);

    let hasRoleInPr = hasRoleInPrBasedOnUserName(pr, userinfo);
    let meetingInfoText = (status, statuses) => {
      if (
        statuses.includes(prStatusEnum.FINALIZED_REVIEWER) &&
        !(status === 'ACCEPTED')
      ) {
        return this.props.intl.formatMessage({
          id: 'prstate.offportalarranged'
        });
      } else {
        switch (status) {
          case 'ACCEPTED':
            return this.props.intl.formatMessage({
              id: 'prstate.agreed'
            });
          case 'DECLINED':
            if (userIsMemberOfHr) {
              return this.props.intl.formatMessage({
                id: 'prstate.cancelled'
              });
            }
            return this.props.intl.formatMessage({
              id: 'prstate.newtermin'
            });
          case 'NO_ANSWER':
            return this.props.intl.formatMessage({
              id: 'prstate.terminsent'
            });
          case 'NOT_REQUESTED':
            if (userIsMemberOfHr) {
              return this.props.intl.formatMessage({
                id: 'prstate.notermin'
              });
            }
            return this.props.intl.formatMessage({
              id: 'prstate.arrangetermin'
            });
          default:
            return this.props.intl.formatMessage({
              id: 'prstate.noinfo'
            });
        }
      }
    };
    let step1 = {
      mainStepLabel: this.props.intl.formatMessage({
        id: 'prstate.preparation'
      }),
      substeps: {
        [prStatusEnum.RELEASED_SHEET_EMPLOYEE]: {
          isCompleted: prStatusesDone[prStatusEnum.RELEASED_SHEET_EMPLOYEE],
          isCurrentUserActionPerformer: hasRoleInPr(['employee']),
          label: `${this.props.intl.formatMessage({
            id: 'prstate.employee'
          })} `,
          rendering: {
            complete: this.props.intl.formatMessage({
              id: 'prstate.finished'
            }),
            incompleteForNonActionPerformer: this.props.intl.formatMessage({
              id: 'prstate.notfinished'
            }),
            incompleteForActionPerformer: this.getIncompleteActionPerformerButton(
              pr,
              prStatusEnum.RELEASED_SHEET_EMPLOYEE
            )
          }
        },
        [prStatusEnum.RELEASED_SHEET_REVIEWER]: {
          isCompleted: prStatusesDone[prStatusEnum.RELEASED_SHEET_REVIEWER],
          isCurrentUserActionPerformer: hasRoleInPr(['supervisor', 'reviewer']),
          label: `${this.props.intl.formatMessage({
            id: 'prstate.reviewer'
          })} `,
          rendering: {
            complete: this.props.intl.formatMessage({
              id: 'prstate.finished'
            }),
            incompleteForNonActionPerformer: this.props.intl.formatMessage({
              id: 'prstate.notfinished'
            }),
            incompleteForActionPerformer: this.getIncompleteActionPerformerButton(
              pr,
              prStatusEnum.RELEASED_SHEET_REVIEWER
            )
          }
        },
        [prStatusEnum.REQUESTED_DATE]: {
          isCompleted: true,
          label: `${this.props.intl.formatMessage({
            id: 'prstate.findtermin'
          })} `,
          rendering: {
            complete: meetingInfoText(meeting.status, pr.statuses)
          }
        }
      }
    };
    let step2 = {
      mainStepLabel: this.props.intl.formatMessage({
        id: 'prstate.termin'
      }),
      substeps: {
        [prStatusEnum.FINALIZED_REVIEWER]: {
          isCompleted: prStatusesDone[prStatusEnum.FINALIZED_REVIEWER],
          isCurrentUserActionPerformer: hasRoleInPr(['supervisor', 'reviewer']),
          label: `${this.props.intl.formatMessage({
            id: 'prstate.reviewer'
          })} `,
          rendering: {
            complete: this.props.intl.formatMessage({
              id: 'prstate.finished'
            }),
            incompleteForNonActionPerformer: this.props.intl.formatMessage({
              id: 'prstate.notfinished'
            }),
            incompleteForActionPerformer: this.getIncompleteActionPerformerButton(
              pr,
              prStatusEnum.FINALIZED_REVIEWER
            )
          }
        }
      }
    };
    let step3 = {
      mainStepLabel: this.props.intl.formatMessage({
        id: 'prstate.conclusion'
      }),
      substeps: {
        [prStatusEnum.FINALIZED_EMPLOYEE]: {
          isCompleted: prStatusesDone[prStatusEnum.FINALIZED_EMPLOYEE],
          isCurrentUserActionPerformer: hasRoleInPr(['employee']),
          label: `${this.props.intl.formatMessage({
            id: 'prstate.employee'
          })} `,
          rendering: {
            complete: this.props.intl.formatMessage({
              id: 'prstate.finished'
            }),
            incompleteForNonActionPerformer: this.props.intl.formatMessage({
              id: 'prstate.notfinished'
            }),
            incompleteForActionPerformer: this.getIncompleteActionPerformerButton(
              pr,
              prStatusEnum.FINALIZED_EMPLOYEE
            )
          }
        }
      }
    };
    let step4 = {
      mainStepLabel: this.props.intl.formatMessage({
        id: 'prstate.postprocessing'
      }),
      substeps: {
        [prStatusEnum.ARCHIVED_HR]: {
          isCompleted: prStatusesDone[prStatusEnum.ARCHIVED_HR],
          isCurrentUserActionPerformer: isHr(this.props.userroles),
          label: `${this.props.intl.formatMessage({
            id: 'prstate.hr'
          })} `,
          rendering: {
            complete: this.props.intl.formatMessage({
              id: 'prstate.archived'
            }),
            incompleteForNonActionPerformer: this.props.intl.formatMessage({
              id: 'prstate.notfinished'
            }),
            incompleteForActionPerformer: this.getIncompleteActionPerformerButton(
              pr,
              prStatusEnum.ARCHIVED_HR
            )
          }
        }
      }
    };

    if (userIsMemberOfHr) {
      return [step1, step2, step3, step4];
    } else {
      return [step1, step2, step3];
    }
  };

  checkRequiredFields = (
    employeeContributionRole,
    employeeContributionLeader,
    overallComment,
    status
  ) => {
    let filledEmployee =
      employeeContributionRole !== null &&
      employeeContributionLeader !== null &&
      employeeContributionRole !== '' &&
      employeeContributionLeader !== '';
    let filledReviewer = overallComment !== null && overallComment !== '';

    let required = { employee: true, reviewer: true };
    switch (status) {
      case prStatusEnum.RELEASED_SHEET_EMPLOYEE:
        return Object.assign({}, required, {
          employee: filledEmployee
        });
      case prStatusEnum.RELEASED_SHEET_REVIEWER:
        return Object.assign({}, required, {
          reviewer: filledReviewer
        });
      case prStatusEnum.FINALIZED_REVIEWER:
        return Object.assign({}, required, {
          reviewer: filledReviewer
        });
      default:
        return required;
    }
  };

  requiredFieldsEmpty = status => {
    let {
      employeeContributionRole,
      employeeContributionLeader,
      overallComment
    } = this.props;

    let fieldFilled = this.checkRequiredFields(
      employeeContributionRole,
      employeeContributionLeader,
      overallComment,
      status
    );

    if (
      prStatusEnum.RELEASED_SHEET_EMPLOYEE === status &&
      false === fieldFilled.employee
    ) {
      this.props.checkRequired(fieldFilled);
      return true;
    } else if (
      prStatusEnum.RELEASED_SHEET_REVIEWER === status &&
      false === fieldFilled.reviewer
    ) {
      this.props.checkRequired(fieldFilled);
      return true;
    } else if (
      prStatusEnum.FINALIZED_REVIEWER === status &&
      false === fieldFilled.reviewer
    ) {
      this.props.checkRequired(fieldFilled);
      return true;
    } else {
      return false;
    }
  };

  getIncompleteActionPerformerButton = (pr, status) => {
    let label =
      status === prStatusEnum.RELEASED_SHEET_REVIEWER ||
      status === prStatusEnum.RELEASED_SHEET_EMPLOYEE
        ? this.props.intl.formatMessage({
            id: 'prstate.release'
          })
        : this.props.intl.formatMessage({
            id: 'prstate.finish'
          });
    return (
      <CheckRequiredClick
        onClick={() => {
          this.props.addPrStatus(pr, status);
        }}
        check={() => !this.requiredFieldsEmpty(status)}
        message={this.props.intl.formatMessage({
          id: 'prstate.fillrequired'
        })}
      >
        <PrStatusActionButton label={label} />
      </CheckRequiredClick>
    );
  };

  render() {
    const { classes, prById, userinfo, meeting } = this.props;

    if (!(prById && meeting)) {
      return null;
    }
    const prStatusesDone = this.getCompletedSubsteps(prById.statuses);

    if (!prStatusesDone) {
      return null;
    }

    const stepStructure = this.updateStepStructure(
      prById,
      userinfo,
      prStatusesDone,
      meeting
    );

    const activeStep = this.calculateActiveStep(prStatusesDone, stepStructure);

    let savingInfo =
      this.props.savingThreads > 0
        ? '...speichert...'
        : 'Alle Änderungen gespeichert.';

    return (
      <Paper className={classes.paper}>
        <List>
          <ListItem>
            <ListItemText>
              {this.props.intl.formatMessage({
                id: 'prstate.progress'
              })}
            </ListItemText>
            <Typography>{savingInfo}</Typography>
          </ListItem>
          <ListItem>
            <PrStatusStepper
              stepStructure={stepStructure}
              activeStep={activeStep}
            />
          </ListItem>
        </List>
      </Paper>
    );
  }
}

export const StyledComponent = withStyles(styles)(PrState);
export default injectIntl(
  connect(
    state => ({
      prById: getPrDetail()(state),
      userroles: getUserroles(state),
      userinfo: getUserinfo(state),
      overallComment: getPrRatings('FULFILLMENT_OF_REQUIREMENT')(state).comment,
      employeeContributionRole: getPrEmployeeContributions(
        'ROLE_AND_PROJECT_ENVIRONMENT'
      )(state).text,
      employeeContributionLeader: getPrEmployeeContributions(
        'INFLUENCE_OF_LEADER_AND_ENVIRONMENT'
      )(state).text,
      requiredFields: getRequiredFields(state),
      meeting: getMeeting(state),
      savingThreads: getSavingThreads(state)
    }),
    {
      checkRequired: changeRequiredFields,
      addPrStatus: actions.addPrStatus
    }
  )(StyledComponent)
);
