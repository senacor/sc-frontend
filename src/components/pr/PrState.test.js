import React from 'react';
import { StyledComponent } from './PrState';
import { createShallow } from '@material-ui/core/test-utils';

describe('PrState Component', () => {
  let shallow = createShallow({ dive: true });

  const prById = {
    id: 1,
    employee: {
      id: 1,
      login: 'lschäfer',
      firstName: 'Lionel',
      lastName: 'Schäfer',
      title: 'DR',
      email: 'lionel.schäfer@senacor.com',
      entryDate: '2004-05-10',
      salaries: [
        {
          workingHoursPerWeek: 40,
          ote: 45000,
          fte: 1,
          basicSalary: 40500,
          validFrom: '2004-05-10'
        },
        {
          workingHoursPerWeek: 40,
          ote: 50000,
          fte: 1,
          basicSalary: 40000,
          validFrom: '2005-05-10'
        }
      ],
      employment: {
        endOfProbationPeriod: '2004-05-10',
        jobGradings: [
          {
            positionId: 1,
            salaryLevel: 1,
            validFrom: '2004-05-09',
            competence: 'DEVELOPMENT'
          },
          {
            positionId: 2,
            salaryLevel: 2,
            validFrom: '2005-05-09',
            competence: 'DEVELOPMENT'
          }
        ],
        leaves: {
          unpaidLeave: [
            {
              from: '2004-05-10',
              to: '2005-03-31'
            }
          ],
          maternityLeave: [
            {
              from: '2004-05-10',
              to: '2004-10-11'
            }
          ],
          parentalLeave: [
            {
              from: '2004-05-10',
              to: '2004-08-28'
            }
          ],
          sabbatical: [
            {
              from: '2004-05-10',
              to: '2004-06-07'
            },
            {
              from: '2004-05-10',
              to: '2005-02-10'
            },
            {
              from: '2004-05-10',
              to: '2005-03-26'
            }
          ]
        }
      }
    },
    supervisor: 'ttran',
    occasion: 'ON_DEMAND',
    status: 'PREPARATION',
    deadline: '2015-05-11',
    _links: {
      self: {
        href: 'http://localhost:8010/api/v1/prs/1'
      }
    }
  };

  let dummyStatusMap = {};

  it('displays the PrState', () => {
    const component = shallow(
      <StyledComponent prById={prById} prStatusesDone={dummyStatusMap} />
    );

    expect(component).toMatchSnapshot();
  });
});
