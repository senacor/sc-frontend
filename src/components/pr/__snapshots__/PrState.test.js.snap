// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrState Component displays the PrState 1`] = `
<WithStyles(Paper)
  className="PrState-paper-1"
>
  <WithStyles(List)>
    <WithStyles(ListItem)>
      <WithStyles(ListItemText)>
        Fortschritt des Reviewprozesses
      </WithStyles(ListItemText)>
    </WithStyles(ListItem)>
    <WithStyles(ListItem)>
      <WithStyles(Stepper)
        activeStep={0}
        className="PrState-stepper-2"
        orientation="vertical"
      >
        <WithStyles(Step)
          key="Vorbereitung PR"
        >
          <WithStyles(StepLabel)>
            Vorbereitung PR
          </WithStyles(StepLabel)>
          <WithStyles(StepContent)>
            <WithStyles(Grid)
              alignItems="center"
              container={true}
              key="SubStepGrid_FILLED_SHEET_EMPLOYEE"
              spacing={8}
            >
              <WithStyles(Grid)
                item={true}
                lg={12}
                md={12}
                sm={12}
                xl={6}
                xs={12}
              >
                <WithStyles(Grid)
                  container={true}
                  direction="row"
                >
                  <WithStyles(Grid)
                    item={true}
                    lg={8}
                    md={6}
                    sm={8}
                    xs={8}
                  >
                    <WithStyles(Typography)>
                      Freigabe Mitarbeiter
                    </WithStyles(Typography)>
                  </WithStyles(Grid)>
                  <WithStyles(Grid)
                    item={true}
                    lg={4}
                    md={6}
                    sm={4}
                    xs={4}
                  />
                </WithStyles(Grid)>
              </WithStyles(Grid)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              alignItems="center"
              container={true}
              key="SubStepGrid_FILLED_SHEET_REVIEWER"
              spacing={8}
            >
              <WithStyles(Grid)
                item={true}
                lg={12}
                md={12}
                sm={12}
                xl={6}
                xs={12}
              >
                <WithStyles(Grid)
                  container={true}
                  direction="row"
                >
                  <WithStyles(Grid)
                    item={true}
                    lg={8}
                    md={6}
                    sm={8}
                    xs={8}
                  >
                    <WithStyles(Typography)>
                      Freigabe Beurteiler
                    </WithStyles(Typography)>
                  </WithStyles(Grid)>
                  <WithStyles(Grid)
                    item={true}
                    lg={4}
                    md={6}
                    sm={4}
                    xs={4}
                  />
                </WithStyles(Grid)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
                lg={12}
                md={12}
                sm={12}
                xl={6}
                xs={12}
              >
                <WithStyles(Button)
                  className="PrState-buttonDesktop-4"
                  onClick={[Function]}
                >
                  PR freigeben
                </WithStyles(Button)>
              </WithStyles(Grid)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              alignItems="center"
              container={true}
              key="SubStepGrid_ALL_DATES_ACCEPTED"
              spacing={8}
            >
              <WithStyles(Grid)
                item={true}
                lg={12}
                md={12}
                sm={12}
                xl={6}
                xs={12}
              >
                <WithStyles(Grid)
                  container={true}
                  direction="row"
                >
                  <WithStyles(Grid)
                    item={true}
                    lg={8}
                    md={6}
                    sm={8}
                    xs={8}
                  >
                    <WithStyles(Typography)>
                      Termin festgesetzt
                    </WithStyles(Typography)>
                  </WithStyles(Grid)>
                  <WithStyles(Grid)
                    item={true}
                    lg={4}
                    md={6}
                    sm={4}
                    xs={4}
                  />
                </WithStyles(Grid)>
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </WithStyles(StepContent)>
        </WithStyles(Step)>
        <WithStyles(Step)
          key="PR abschließen (Beurteiler)"
        >
          <WithStyles(StepLabel)>
            PR abschließen (Beurteiler)
          </WithStyles(StepLabel)>
          <WithStyles(StepContent) />
        </WithStyles(Step)>
        <WithStyles(Step)
          key="PR abschließen (Mitarbeiter)"
        >
          <WithStyles(StepLabel)>
            PR abschließen (Mitarbeiter)
          </WithStyles(StepLabel)>
          <WithStyles(StepContent) />
        </WithStyles(Step)>
        <WithStyles(Step)
          key="PR archivieren (HR)"
        >
          <WithStyles(StepLabel)>
            PR archivieren (HR)
          </WithStyles(StepLabel)>
          <WithStyles(StepContent) />
        </WithStyles(Step)>
      </WithStyles(Stepper)>
    </WithStyles(ListItem)>
  </WithStyles(List)>
</WithStyles(Paper)>
`;
